src/stieltjes.jl

```diff
@@ -75,4 +75,119 @@ end
     D = T \ apply(*, log.(abs.(x .- x')), P)
     c = inv(2*kr.A)
     T[kr,:] * Diagonal(Vcat(2*convert(V,π)*c*log(c), 2c*D.diag.args[2]))
+end
+
+#################################################
+# ∫f(x)g(x)(t-x)^a dx evaluation where f and g in Legendre
+#################################################
+const PowKernelPoint{T,V,D,F} =  BroadcastQuasiVector{T, typeof(^), Tuple{ContinuumArrays.AffineQuasiVector{T, V, Inclusion{V, D}, T}, F}}
+
+struct PowerLawMatrix{T} #<: AbstractMatrix{T} # TODO
+    a::T
+    t::T
+end
+
+############
+# METHODS
+############
+function dot(f::AbstractVector{T},K::PowKernelPoint{Float64,Float64,ChebyshevInterval{Float64}},g::AbstractVector{T}) where T
+    (lf, lg) = (length(f),length(g))
+    a = K.args[2]
+    t = (K.args[1])[0.] # there must be something better than this? 
+                        # maybe something in the spirit of (K.args[1]).args[1]?
+    t<1 && error("t must be greater than 1.")
+    (lf<∞) && (lg<∞) && return pointwisedot(f,g,a,t)
+    ((lf<∞) || (lg<∞)) && error("TODO: Currently only both finite or both infinite.")
+    # seek naive convergence for infinite input.
+    (i,conv1,conv2) = (0,1,2)
+    while abs(conv1-conv2)>1e-12
+        i = i+1
+        conv1 = pointwisedot(f[1:i*20],g[1:i*20],a,t)
+        conv2 = pointwisedot(f[1:2*i*20],g[1:2*i*20],a,t)
+    end
+    return conv2
+end
+function dot(f::AbstractVector,M::PowerLawMatrix,g::AbstractVector)
+    return dot(f,(M.t .- axes(Legendre(),1)).^M.a,g)
+end
+function *(P::Legendre,M::PowerLawMatrix)
+    return (M.t .- axes(P,1)).^M.a.*P
+end
+function *(g::Adjoint,K::PowKernelPoint{<:Any,<:Any,<:ChebyshevInterval},f::AbstractVector)
+    return dot(g',K,f)
+end
+
+############
+# EVALUATE RECURRENCE
+############
+# experimental pointwise power law integral of Legendre product by recurrence
+function pointwisedot(f::AbstractVector{T},g::AbstractVector{T},a::Real,t::Real) where T
```

Comment:
I'm a bit confused here. What is `coeff`? Can he adaptivity be done inside here? That is, let `f` and `g` be infinite.

Reply:
With the way this currently stands (which I expect will change substantially) adaptivity can only work outside of that function by supplying larger pieces of f and g, i.e. overload dot(f,K,g) to do the adaptivity part when f and g are infinite. 

Is there a more natural way to have an adaptive scalar? That is, since dot(f,K,g) is the integral and returns a scalar value, what exactly would it mean to give adaptive output here? Is there a comparable example somewhere that I can have a look at?